nautobot/extras/api/views.py

```diff
@@ -522,16 +522,28 @@ def _run_job(request, job_model, legacy_response=False):
         raise MethodNotAllowed(request.method, detail="This job's source code could not be located and cannot be run")
     job = job_class()
 
-    input_serializer = serializers.JobInputSerializer(data=request.data, context={"request": request})
+    data = request.data
+    files = None
+
+    if "multipart/form-data" in request.content_type:
+        data = request._data  # .data will return data and files, we just want the data
+        files = request.FILES
+
+    input_serializer = serializers.JobInputSerializer(data=data, context={"request": request})
     input_serializer.is_valid(raise_exception=True)
 
     data = input_serializer.validated_data.get("data", {})
     commit = input_serializer.validated_data.get("commit", None)
     if commit is None:
         commit = job_model.commit_default
 
+    cleaned_data = None
     try:
-        job.validate_data(data)
+        cleaned_data = job.validate_data(data, files)
+        cleaned_data.pop(
+            "_commit"
+        )  # We don't get commit from the form, instead it's part of the serializer's validated data
```

Comment:
https://github.com/nautobot/nautobot/blob/v1.4.7/nautobot/extras/views.py#L1139

Reply:
You might want this to be a `data.pop("_commit", None)` unless there's an assertion this will always be there, it could result in a `KeyError`.